server/images.go

```diff
@@ -294,6 +297,40 @@ func CreateModel(name string, path string, fn func(resp api.ProgressResponse)) e
 					layers = append(layers, newLayer)
 				}
 			}
+		case "adapter":
```

Comment:
A lot of this code is copied from the model code path. #288 refactors this into a separate function